compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt

```diff
@@ -110,7 +110,9 @@ class AdaptersProcessingStep(
                 .superclass(getAdapterType(typeName))
                 .addField(optionsField)
                 .addFields(generateFields(adapterKeys))
-                .addMethod(generateConstructor(adapterKeys, genericTypes))
+                .applyIf(adapterKeys.isNotEmpty()) {
```

Comment:
The adapter keys will be non empty when the class is a generic class, right?

Reply:
Yes. The use of a generic requires an adapter, and elsewhere the adapter list is used as the sole source of information for pulling out that generic information (up in the factory generator I think).

Reply:
Cool.